pennylane/qaoa/cost.py

```diff
@@ -506,7 +572,7 @@ def max_weight_cycle(graph, constrained=True):
     our subset of edges composes a `cycle <https://en.wikipedia.org/wiki/Cycle_(graph_theory)>`__.
 
     Args:
-        graph (nx.Graph): the directed graph on which the Hamiltonians are defined
+        graph (nx.Graph or rx.Py(Di)Graph): the directed graph on which the Hamiltonians are defined
```

Comment:
```suggestion
        graph (nx.Graph or rx.PyGraph or rx.PyDiGraph): the directed graph on which the Hamiltonians are defined
```

Reply:
Thanks! Fixed in all places :slightly_smiling_face: